src/api/worker/rest/EntityRestCache.ts

```diff
@@ -255,110 +243,186 @@ export class EntityRestCache implements IEntityRestCache {
+		start: Id,
+		count: number,
+		reverse: boolean
+	) {
+		// Create a new range and load everything
+		const entities = await this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
+
+		// Initialize a new range for this list
+		await this.storage.setNewRangeForList(typeRef, listId, start, start)
+
+		// The range bounds will be updated in here
+		await this.updateRangeInStorage(typeRef, listId, count, reverse, entities)
+	}
 
-				if (isLoadingAwayFromRange) {
-					// Start is outside the range, and we are loading away from the range, so we grow until we are able to provide enough
-					// entities starting at startId
+	/**
+	 * Returns part of a request from the cache, and the remainder is loaded from the server
+	 * range:          |-----------|
+	 * request:               *----------->
+	 * range becomes: |-------------------|
+	 */
+	private async extendFromWithinRange<T extends ListElementEntity>(typeRef: TypeRef<T>, listId: Id, start: Id, count: number, reverse: boolean) {
 
-					// Which end of the range to start loading from
-					const loadStartId = reverse
-						? range.lower
-						: range.upper
+		const {newStart, newCount} = await this.recalculateRangeRequest(typeRef, listId, start, count, reverse)
+		if (newCount > 0) {
+			// We will be able to provide some entities from the cache, so we just want to load the remaining entities from the server
+			const entities = await this.entityRestClient.loadRange(typeRef, listId, newStart, newCount, reverse)
+			await this.updateRangeInStorage(typeRef, listId, newCount, reverse, entities)
+		}
+	}
 
-					// Load from the end of the range in the direction of the startId. then, if all available elements have been loaded or the requested number is in cache, return from cache. otherwise load again the same way.
-					const entities = await this.entityRestClient.loadRange(typeRef, listId, loadStartId, count, reverse)
+	/**
+	 * Start was outside the range, and we are loading away from the range
+	 * Keeps loading elements from the end of the range in the direction of the startId.
+	 * Returns once all available elements have been loaded or the requested number is in cache
+	 * range:		   |-----------------|
+	 * request:							    *------->
+	 * range becomes:  |----------------------------|
+	 */
+	private async extendAwayFromRange<T extends ListElementEntity>(
+		typeRef: TypeRef<T>,
+		listId: Id,
+		start: Id,
+		count: number,
+		reverse: boolean
+	) {
+		// Start is outside the range, and we are loading away from the range, so we grow until we are able to provide enough
+		// entities starting at startId
+		while (true) {
+			const range = assertNotNull(await this.storage.getRangeForList(typeRef, listId))
+
+			// Which end of the range to start loading from
+			const loadStartId = reverse
+				? range.lower
+				: range.upper
+
+			// Load some entities
+			const entities = await this.entityRestClient.loadRange(typeRef, listId, loadStartId, count, reverse)
```

Comment:
We can actually be smarter here and load more than count if it makes sense. Otherwise this might lead to a situation where you want to load 1 element far outside of range and we will keep loading in small chunks.

I don't think it's absolutely necessary now but it can absolutely happen.

Maybe easy solution would be to do something like `Math.max(count, DEFAULT_CHUNK_SIZE)`